src/ol/renderer/canvas/VectorTileLayer.js

```diff
@@ -604,6 +604,7 @@ class CanvasVectorTileLayerRenderer extends CanvasTileLayerRenderer {
    * @param {import("../../PluggableMap.js").FrameState} frameState Frame state.
    */
   renderDeclutter(frameState) {
+    this.getLayer().getOpacity();
```

Comment:
This does nothing, it should set and reset globalAlpha in the same way as in the renderFrame method.

Reply:
Thanks, I had forgotten to finish that. I just added another commit with a test for decluttering.

Regarding your question why the vector mode example is darker: this is because overlapping geometries are each drawn with the layer opacity, so the overall opacity will be higher. For vector layers, this was solved with quite an expensive workaround (render to a separate canvas, and dim that down with the layer opacity - see #6981), but I think this is a corner case and does not justify the extra rendering work.